UI/uiModelComparison.py

```diff
@@ -0,0 +1,62 @@
+'''
+file implementing the UI of model performance comparison of an image
+'''
+
+# --- /
+# -- / external imports
+import streamlit as st
+import PIL
+
+
+def compareModels(confidenceThreshold:float):
+    '''
+    provides the UI for uploading an image, choosing a model 
+    and starting the detection at the press of an button. 
+    '''
+    # TODO may become redundant when refactoring is finished
+    imgFile = None
+    imgBytes = st.file_uploader("Upload an image", type=['png', 'jpeg', 'jpg'])
+    if imgBytes:
+        imgFile = PIL.Image.open(imgBytes)
+        st.image(imgFile, caption="Image")
+
+    modelOptions = ["YOLOv8n", "YOLOv8s",
+                    "YOLOv8m", "YOLOv8l", "Custom", "SSD"]
+    modelOne = st.selectbox("Select first model", modelOptions)
+    modelTwo = st.selectbox("Select second model", modelOptions)
+
+    if st.button("Start Comparison"):
+        if imgFile:
+            metaData = detectObjects(imgFile, modelOne, modelTwo, confidenceThreshold)
+            compareObjects(metaData)
+        else:
+            st.error("No image provided")
+
+
+# TODO not sure if the types are correct
+def detectObjects(image:PIL.Image, modelOne:str, modelTwo:str, confidenceThreshold:float) -> str:
```

Comment:
Looking good, however this wont scale well with multiple models to compare with. One could add a better implementation later, I think yours is fine for the current process too